huaweicloud/services/hss/resource_huaweicloud_hss_cce_protection.go

```diff
@@ -0,0 +1,569 @@
+	}
+
+	return bodyParam
+}
+
+func updateCCEProtection(client *golangsdk.ServiceClient, d *schema.ResourceData, cfg *config.Config) error {
+	var (
+		httpUrl   = "v5/{project_id}/container/kubernetes/clusters/{cluster_id}/daemonsets"
+		epsID     = cfg.GetEnterpriseProjectID(d)
+		clusterID = d.Get("cluster_id").(string)
+		region    = cfg.GetRegion(d)
+	)
+
+	requestPath := client.Endpoint + httpUrl
+	requestPath = strings.ReplaceAll(requestPath, "{project_id}", client.ProjectID)
+	requestPath = strings.ReplaceAll(requestPath, "{cluster_id}", clusterID)
+	if epsID != "" {
+		requestPath += "?enterprise_project_id=" + epsID
+	}
+
+	requestOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		MoreHeaders: map[string]string{
+			"Content-Type": "application/json",
+			"region":       region,
+		},
+		JSONBody: utils.RemoveNil(buildUpdateCCEProtectionBodyParam(d)),
+	}
+
+	_, err := client.Request("PUT", requestPath, &requestOpt)
+	return err
+}
+
+func resourceCCEProtectionCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var (
+		cfg       = meta.(*config.Config)
+		region    = cfg.GetRegion(d)
+		product   = "hss"
+		clusterID = d.Get("cluster_id").(string)
+	)
+
+	client, err := cfg.NewServiceClient(product, region)
+	if err != nil {
+		return diag.Errorf("error creating HSS client: %s", err)
+	}
+
+	if err := createCCEProtection(client, d, cfg); err != nil {
+		return diag.Errorf("error opening HSS protection for CCE (%s): %s", clusterID, err)
+	}
+	d.SetId(clusterID)
+
+	if err := waitingForProtectStatusProtecting(ctx, client, d, cfg, d.Timeout(schema.TimeoutCreate)); err != nil {
+		return diag.Errorf("error waiting for CCE (%s) protection configuration open to complete: %s", clusterID, err)
+	}
+
+	if err := updateCCEProtection(client, d, cfg); err != nil {
+		return diag.Errorf("error configuring HSS protection configuration for CCE (%s) in"+
+			" creation operation: %s", clusterID, err)
+	}
+
+	return resourceCCEProtectionRead(ctx, d, meta)
+}
+
+func flattenPreferPacketCycle(clusterRespBody interface{}) bool {
+	preferPacketCycle := int(utils.PathSearch("prefer_packet_cycle", clusterRespBody, float64(0)).(float64))
+	return preferPacketCycle != 0
+}
+
+func resourceCCEProtectionRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var (
+		cfg         = meta.(*config.Config)
+		region      = cfg.GetRegion(d)
+		product     = "hss"
+		clusterID   = d.Get("cluster_id").(string)
+		clusterName = d.Get("cluster_name").(string)
+		epsID       = cfg.GetEnterpriseProjectID(d)
+	)
+
+	client, err := cfg.NewServiceClient(product, region)
+	if err != nil {
+		return diag.Errorf("error creating HSS client: %s", err)
+	}
+
+	respBody, err := ReadCCEProtection(client, clusterID, clusterName, epsID, region)
+	if err != nil {
+		return diag.Errorf("error retrieving CCE protection configuration: %s", err)
```

Comment:
```suggestion
		return diag.Errorf("error retrieving HSS protection configuration for CCE: %s", err)
```

Reply:
There are similar issues in other places as well

Reply:
OK